### PR TITLE
CHORE: expose show for plotting

### PR DESCRIPTION
**Pull request by dcrawforAtAnsys**
Expose the keyword argument "show" for all methods that create matplotlib figures. Default behavior is unchanged. Plots will be rendered when the method is called.

### CODE REVIEW

File: pyaedt/generic/plot.py [hunk 1]
```diff
@@ -542,7 +550,8 @@ def plot_matplotlib(
         List of annotations to add to the plot. The format is [x, y, string, dictionary of font options].
         Default is `None`.
```
Comment (PipKat) [inline pyaedt/generic/plot.py:551]:
```suggestion
        The default is ``None``.
```

File: pyaedt/generic/plot.py [hunk 2]
```diff
@@ -638,8 +647,8 @@ def plot_contour(
     snapshot_path : str, optional
         Full path to save the image save. The default is ``None``.
```
Comment (PipKat) [inline pyaedt/generic/plot.py:648]:
```suggestion
        Full path to save the image to. The default is ``None``.
```

File: pyaedt/modules/solutions.py [hunk 3]
```diff
@@ -778,6 +778,7 @@ def plot(
         title="",
         snapshot_path=None,
         is_polar=False,
+        show=True,
     ):
         """Create a matplotlib plot based on a list of data.
```
Comment (PipKat) [inline pyaedt/modules/solutions.py:783]:
```suggestion
        """Create a Matplotlib plot based on a list of data.
```

File: pyaedt/modules/solutions.py [hunk 4]
```diff
@@ -808,6 +809,9 @@ def plot(
             Full path to image file if a snapshot is needed.
```
Comment (PipKat) [inline pyaedt/modules/solutions.py:809]:
```suggestion
            Full path to the image file if a snapshot is needed.
```

File: pyaedt/modules/solutions.py [hunk 5]
```diff
@@ -808,6 +809,9 @@ def plot(
...
         is_polar : bool, optional
             Set to `True` if this is a polar plot.
```
Comment (PipKat) [inline pyaedt/modules/solutions.py:811]:
```suggestion
            Whether this is a polar plot. The default is ``False``.
```

File: pyaedt/modules/solutions.py [hunk 6]
```diff
@@ -868,6 +872,7 @@ def plot_3d(
         formula=None,
         size=(2000, 1000),
         snapshot_path=None,
+        show=True,
     ):
         """Create a matplotlib 3d plot based on a list of data.
```
Comment (PipKat) [inline pyaedt/modules/solutions.py:877]:
```suggestion
        """Create a Matplotlib 3D plot based on a list of data.
```

Comment (dcrawforAtAnsys) [reply pyaedt/modules/solutions.py:877]: Added all suggestions from @PipKat .

File: pyaedt/generic/plot.py [hunk 7]
```diff
@@ -378,14 +378,14 @@ def plot_polar_chart(
     fig.set_size_inches(size[0] / dpi, size[1] / dpi)
     if snapshot_path:
         fig.savefig(snapshot_path)
-    if show:
+    if show and not is_notebook():
```
Comment (SMoraisAnsys) [inline pyaedt/generic/plot.py:381]:
```suggestion
    if show:
```

Comment (dcrawforAtAnsys) [reply pyaedt/generic/plot.py]: @SMoraisAnsys Thanks. I missed that one.

**Review by SMoraisAnsys — COMMENTED** [review]
To extend your changes and be consistent, should we also update `plot` from `pyaedt.generic.plot.ModelPlotter` ? The current code is

```python
        if export_image_path:
            self.pv.show(screenshot=export_image_path, full_screen=True)
        elif show and self.is_notebook:  # pragma: no cover
            self.pv.show()  # pragma: no cover
        elif show:
            self.pv.show(full_screen=True)  # pragma: no cover
```

Since we are starting to follow the approach of allowing both writting a figure in a file AND plotting it, I would update the previous snipper into

```python
        if export_image_path:
            self.pv.show(screenshot=export_image_path, full_screen=True)
        if show:  # pragma: no cover
            self.pv.show(full_screen=not self.is_notebook)
```

What do you think ?

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, I just left a minor comment

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**